src/rest_api/api/search/superstructure_search.rs

```diff
@@ -0,0 +1,177 @@
+            match scaffold_matches {
+                Ok(scaffold_matches) => Some(scaffold_matches),
+                Err(e) => {
+                    return GetStructureSearchResponse::Err(Json(StructureResponseError {
+                        error: e.to_string(),
+                    }))
+                }
+            }
+        }
+        None => None,
+    };
+
+    let results = superstructure_search(
+        &searcher,
+        &query_canon_taut,
+        &matching_scaffolds,
+        fingerprint.0.as_bitslice(),
+        &descriptors,
+        result_limit,
+        extra_query,
+    );
+
+    let mut results = match results {
+        Ok(results) => results,
+        Err(e) => {
+            return GetStructureSearchResponse::Err(Json(StructureResponseError {
+                error: e.to_string(),
+            }))
+        }
+    };
+
+    let mut used_tautomers = false;
+    let mut num_tauts_used = 0;
+    if !results.is_empty() {
+        num_tauts_used = 1;
+    }
+
+    if results.len() < result_limit {
+        let tautomers = get_tautomers(&query_canon_taut);
+
+        if tautomers.len() > 1 && tautomer_limit > 1 {
+            for test_taut in tautomers {
+                // don't reuse the canonical tautomer
+                if test_taut.as_smiles() == query_canon_taut.as_smiles() {
+                    continue;
+                }
+
+                let taut_attributes = get_cpd_properties(&test_taut);
+
+                let taut_attributes = match taut_attributes {
+                    Ok(taut_attributes) => taut_attributes,
+                    Err(_) => continue,
+                };
+
+                let (taut_fingerprint, taut_descriptors) = taut_attributes;
+
+                let matching_scaffolds = match scaffolds {
+                    Some(scaffolds) => {
+                        let scaffold_matches = scaffold_search(&test_taut, scaffolds);
+                        match scaffold_matches {
+                            Ok(scaffold_matches) => Some(scaffold_matches),
+                            Err(e) => {
+                                return GetStructureSearchResponse::Err(Json(
+                                    StructureResponseError {
+                                        error: e.to_string(),
+                                    },
+                                ))
+                            }
+                        }
+                    }
+                    None => None,
+                };
+
+                let taut_results = superstructure_search(
+                    &searcher,
+                    &test_taut,
+                    &matching_scaffolds,
+                    taut_fingerprint.0.as_bitslice(),
+                    &taut_descriptors,
+                    result_limit,
+                    extra_query,
+                );
+
+                let taut_results = match taut_results {
+                    Ok(taut_results) => taut_results,
+                    Err(_) => continue,
```

Comment:
Anytime I see `continue` I think of a chunk of work that could happen inside of `filter_map`.

Reply:
Would a `filter_map` work considering the explanation above?

Reply:
Alright, I've incorporated some `filter_map` usages.